pkg/cmd/cli/cmd/logs.go

```diff
@@ -189,5 +218,43 @@ func (o OpenShiftLogsOptions) RunLog() error {
 		// Use our own options object.
 		o.KubeLogOptions.Options = o.Options
 	}
-	return o.KubeLogOptions.RunLogs()
+	if !o.JenkinsLogs {
+		return o.KubeLogOptions.RunLogs()
+	}
+
+	bc, ok := o.KubeLogOptions.Object.(*buildapi.BuildConfig)
+	var urlString, buildName string
+	if !ok {
+		build, ok := o.KubeLogOptions.Object.(*buildapi.Build)
+		if ok {
+			buildName = build.ObjectMeta.Name
+			urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+			if len(urlString) == 0 {
+				return errors.New(fmt.Sprintf("The build %s did not have the correct log URL", buildName))
+			}
+		}
+	} else {
+		buildName = buildutil.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
+		build, _ := o.Client.Builds(o.Namespace).Get(buildName, metav1.GetOptions{})
+		if build == nil {
+			return errors.New(fmt.Sprintf("The build %s for build config %s was not found", buildName, bc.ObjectMeta.Name))
+		}
+		urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+		if len(urlString) == 0 {
+			return errors.New(fmt.Sprintf("The build %s did not have the correct log URL", buildName))
+		}
+	}
+	if !ok {
```

Comment:
You shadowed "ok" above, i'm not sure this check represents what you want it to.

if the buildconfig check fails, ok will be false, then the build check can succeed, but the "ok" there is shadowed, so the outer ok remains false.  So in the case that you have a build object w/ no valid url, you're going to fall through to this point and get this type error reported.


https://play.golang.org/p/RvuZP_p9Ir

(it'd also make the code easier to follow if the variables were named like "isBC" and "isBuild").

if might be easiest to just check both cases up front, report an error if it's not a BC and not a Build, and then proceed into the rest of the logic already knowing the object type.